### ISSUE TITLE
PageIterator TypeDefs missing methods

### ISSUE THREAD

**Issue by deric4** [vemel/mypy_boto3_builder#106]
**Describe the bug**
`PageIterators` don't seem to inherit all of the methods from `botocore.paginate.PageIterators`
**To Reproduce**
1. Install 'boto3-stubs[...]'
2.  Run mypy on the following code sample

```python
import boto3
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from mypy_boto3_ec2.client import EC2Client
    from mypy_boto3_ec2.service_resource import EC2ServiceResource
else:
    EC2ServiceResource = object
    EC2Client = object

ec2_resource: EC2ServiceResource = boto3.resource("ec2")
ec2_client: EC2Client = ec2_resource.meta.client

result = ec2_client.get_paginator("describe_instances").paginate().build_full_result()
```
3. Actual mypy output

```shell-session
$ mypy --pretty test.py           
test.py:14: error: "Iterator[DescribeInstancesResultTypeDef]" has no attribute "build_full_result"
    result = ec2_client.get_paginator("describe_instances").paginate().build_full_result()
             ^
Found 1 error in 1 file (checked 1 source file)
```

**Expected mypy-output**

```shell-session
$ mypy test.py
Success: no issues found in 1 source file
```

**Additional context**
Your OS, `boto3-stubs` installation method, `boto3` version, etc.

```shell-session
$ sw_vers
ProductName:    macOS
ProductVersion: 11.2.3
BuildVersion:   20D91
```

### Installation Method
`pip`

### Versions
```
boto3                      1.18.19
boto3-stubs                1.18.19
botocore                   1.21.19
botocore-stubs             1.21.19
mypy                       0.910
mypy-boto3-builder         5.1.0
mypy-boto3-cloudformation  1.18.19
mypy-boto3-dynamodb        1.18.19
mypy-boto3-ec2             1.18.19
mypy-boto3-lambda          1.18.19
mypy-boto3-rds             1.18.19
mypy-boto3-s3              1.18.19
mypy-boto3-sqs             1.18.19
mypy-boto3-synthetics      1.18.19
```

### References
- https://github.com/boto/botocore/blob/develop/tests/unit/test_paginate.py

**Comment by vemel** [comment]
Hello!

This fix took a while because Subscripts cannot be used for non-subscriptable classes, like `PageIterator`. However, starting from the next `boto3-stubs` release, all PageIterators will have annotations for all methods, and the iterable type is still annotated properly.

The fix is a bit hacky, but it works. 

Thank you for your report and patience!

**Comment by deric4** [comment]
@vemel 

Awesome! Thank you for all the work you put into the project :D

**Comment by vemel** [comment]
You are welcome! Let me know if you find other bugs or have any suggestions for improvement!

**Comment by omBratteng** [comment]
@vemel not sure if it's a bug or not, but I get `"search" of "PageIterator" does not return a value`. I think it comes from this https://github.com/vemel/mypy_boto3_builder/blob/master/mypy_boto3_builder/botocore_stubs_static/paginate.pyi#L41
![image](https://user-images.githubusercontent.com/1681525/140154866-5301c1ac-8ab1-4913-8f53-3f36d41c3c79.png)
